### PR TITLE
Remove multihash from Swarm bzz:// for Feeds

### PR DESCRIPTION
**Pull request by jpeletier**
This PR removes the required multihash encoding for feed updates in combination with `bzz:`. Prior to this change, users had to prepend `0x1b20` multihash header to feed updates containing pointers to content. This is no longer necessary: To use Feeds + bzz, only the 32-byte Swarm hash of the manifest needs to be included in the feed update.

Additionally, this PR adds:
* Missing test case in client tests regarding `bzz:` in combination with feeds
* Extensive comments to better document tests, including renaming variables to make the tests easier to follow.
* Remove the now unused `swarm/multihash` package.

### CODE REVIEW

**Comment by justelad** [comment]
@jpeletier there are still references to `multihash`:
```
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:19:2: cannot find package "github.com/ethereum/go-ethereum/swarm/multihash" in any of:
	/Users/travis/gopath/src/github.com/ethereum/go-ethereum/vendor/github.com/ethereum/go-ethereum/swarm/multihash (vendor tree)
	/Users/travis/.gimme/versions/go1.11.2.darwin.amd64/src/github.com/ethereum/go-ethereum/swarm/multihash (from $GOROOT)
	/Users/travis/gopath/src/github.com/ethereum/go-ethereum/swarm/multihash (from $GOPATH)
```

File: swarm/api/http/server_test.go [hunk 1]
```diff
@@ -69,98 +68,130 @@ func newTestSigner() (*feed.GenericSigner, error) {
...
+	feedUpdateURL, err := url.Parse(fmt.Sprintf("%s/bzz-feed:/", srv.URL))
 	if err != nil {
 		t.Fatal(err)
 	}
-	query := testUrl.Query()
+	query := feedUpdateURL.Query()
 	body := updateRequest.AppendValues(query) // this adds all query parameters and returns the data to be posted
-	query.Set("manifest", "1")                // indicate we want a manifest back
-	testUrl.RawQuery = query.Encode()
+	query.Set("manifest", "1")                // indicate we want a feed manifest back
```
Comment (nolash) [inline swarm/api/http/server_test.go]: This is still to be deprecated, right? Maybe add a TODO comment.

**Comment by nolash** [comment]
To follow up on @justelad comment:

```
$ grep -i multihash cmd/swarm/ -r
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:	"github.com/ethereum/go-ethereum/swarm/multihash"
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:	multihashHex := hexutil.Encode(multihash.ToMultihash(hashBytes))
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:	cmd = exec.Command("swarm", "--bzzaccount", pkFile.Name(), "--bzzapi", endpoints[0], "feed", "update", "--topic", topicHex, multihashHex)
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:	cmd = exec.Command("swarm", "--bzzaccount", pkFile.Name(), "--bzzapi", endpoints[0], "feed", "update", "--name", subTopicHex, multihashHex)
cmd/swarm/swarm-smoke/feed_upload_and_sync.go:	cmd = exec.Command("swarm", "--bzzaccount", pkFile.Name(), "--bzzapi", endpoints[0], "feed", "update", "--topic", topicHex, "--name", subTopicHex, multihashHex)
cmd/swarm/feeds.go:// swarm feed create <frequency> [--name <name>] [--data <0x Hexdata> [--multihash=false]]
cmd/swarm/feeds.go:// swarm feed update <Manifest Address or ENS domain> <0x Hexdata> [--multihash=false]
```

File: swarm/api/http/server_test.go (hunk 1 above)
Comment (jpeletier) [reply swarm/api/http/server_test.go]: There would be TODOs in more places. I would have it as out of scope for this PR.

**Comment by jpeletier** [comment]
@justelad @nolash - removed multihash also from the smoke test.
Note that smoke tests won't work until the clusters are upgraded to this PR.

**Pull request merged**